mission_control/test/test_waypoint_behavioral.py

```diff
@@ -0,0 +1,166 @@
+
+        simulated_mission_control_load_mission_pub = rospy.Publisher('/mission_control_node/load_mission',
+                                                                     LoadMission, latch=True, queue_size=1)
+
+        simulated_mission_control_execute_mission_pub = rospy.Publisher(
+            '/mission_control_node/execute_mission', ExecuteMission, latch=True, queue_size=1)
+
+        simulated_abort_mission_msg_pub = rospy.Publisher(
+            '/mission_control_node/abort_mission', AbortMission, latch=True, queue_size=1)
+
+        simulated_query_mission_msg_pub = rospy.Publisher(
+            '/mission_control_node/query_missions', QueryMissions, latch=True, queue_size=1)
+
+        simulated_remove_mission_msg_pub = rospy.Publisher(
+            '/mission_control_node/remove_missions', RemoveMissions, latch=True, queue_size=1)
+
+        simulated_mission_manager_execute_mission_pub = rospy.Publisher('/mission_control_node/execute_mission',
+                                                                        ExecuteMission, queue_size=1)
+
+        simulated_pose_estimator_pub = rospy.Publisher('/pose/corrected_data',
+                                                       CorrectedData, queue_size=1)
+
+        # Load mission with waypoint behavioral
+        self.mission_load_state_flag = False
+        mission_to_load = LoadMission()
+        mission_to_load.mission_file_full_path = self.dir_path + \
+            "test_missions/waypoint_mission_test.xml"
+        simulated_mission_control_load_mission_pub.publish(mission_to_load)
+        while not rospy.is_shutdown() and not self.mission_load_state_flag:
+            rospy.sleep(0.1)
+        self.assertEqual(self.mission_load_state,
+                         ReportLoadMissionState.SUCCESS)
+
+        # TEST - Query Mission
+        mission_to_query = QueryMissions()
+        simulated_query_mission_msg_pub.publish(mission_to_query)
+        while (not rospy.is_shutdown() and self.report_mission_flag == False):
+            rospy.sleep(0.1)
+        self.assertEqual(
+            self.report_mission.missions[0].mission_description, "mission test")
+
+        # TEST - Execute Mission
+        execute_mission_command = ExecuteMission()
+        execute_mission_command.mission_id = 1
+        simulated_mission_manager_execute_mission_pub.publish(
+            execute_mission_command)
+        rospy.sleep(1)
+
+        # TEST - Waypoint has published the goal
+        while (not rospy.is_shutdown() and self.waypoint_goal_flag == False):
+            rospy.sleep(0.1)
+        self.assertEqual(self.waypoint_goal, [3.0, 4.0, 1.0, 0.0, 6.0, 4])
+        rospy.sleep(4)
+
+        # TEST - Pusblish Pose Estimator Data
+        pose_estimator_corrected_data = CorrectedData()
+        rospy.loginfo(pose_estimator_corrected_data)
+        simulated_pose_estimator_pub.publish(pose_estimator_corrected_data)
+
+        # TODO
+        # test if the behavior has ended.
```

Comment:
@patapufate same comments as in `test_jaus_ros_bridge_interface_mission_control.py` apply here.

Reply:
Put it in standby